Source/Renderer/GLSLModernizer.js

```diff
@@ -0,0 +1,336 @@
+define([
+        '../Core/DeveloperError',
+    ], function(
+        DeveloperError) {
+    'use strict';
+
+    /**
+     * A function to port GLSL shaders from GLSL ES 1.00 to GLSL ES 2.00
+     *
+     * This function is nowhere near comprehensive or complete. It just
+     * handles some common cases.
+     *
+     * @private
+     */
+    function glslModernizeShaderSource(shaderSource, isFragmentShader) {
+        for (var i = 0; i < shaderSource.sources.length; i++) {
+            shaderSource.sources[i] = glslModernizeShaderText(shaderSource.sources[i], isFragmentShader, i === 0);
+        }
+    }
+
+    // Note that this function requires the presence of the
+    // "#define OUTPUT_DECLARATION" line that is appended
+    // by ShaderSource
+    function glslModernizeShaderText(source, isFragmentShader, first) {
+        var mainFunctionRegex = /void\s+main\s*\((void)?\)/;
+        var outputDeclarationRegex = /#define OUTPUT_DECLARATION/;
+        var splitSource = source.split('\n');
+
+        if (source.search(/#version 300 es/g) !== -1) {
+            return source;
+        }
+        
+        var outputDeclarationLine = -1;
+        var number;
+        for (number = 0; number < splitSource.length; ++number) {
+            var line = splitSource[number];
+            if (outputDeclarationRegex.exec(line)) {
+                outputDeclarationLine = number;
+                break;
+            }
+        };
+        
+        if (outputDeclarationLine === -1) {
+            for (number = 0; number < splitSource.length; ++number) {
+                var line = splitSource[number];
+                if (mainFunctionRegex.exec(line)) {
+                    outputDeclarationLine = number;
+                }
+            };
+        }
+
+        if (outputDeclarationLine === -1) {
+            throw new DeveloperError('Could not find a #define OUTPUT_DECLARATION nor a main function!');
+        }
```

Comment:
If GLSLModernizer is only going to be used by ShaderSource, then it makes sense to assume that OUTPUT_DECLARATION would always be there. The "check for main" part is a fallback if it isn't. 

Given that we're relying ShaderSource to pass flags in for us, we can remove that check.